scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,191 @@
+
+    args = parser.parse_args()
+
+    return(args)
+
+def setup_logging(debug):
+    """Sets up the logging module and logging level."""
+
+    #Use capgen logging tools
+    logger = init_log('ccpp_track_variables')
+
+    if debug:
+        set_log_level(logger, logging.DEBUG)
+        logger.info('Logging level set to DEBUG')
+    else:
+        set_log_level(logger, logging.WARNING)
+    return logger
+
+def parse_suite(sdf, run_env):
+    """Reads the provided sdf, parses into a Suite data structure, including the "call tree":
+       the ordered list of schemes for the suite specified by the provided sdf"""
+    run_env.logger.info(f'Reading sdf {sdf} and populating Suite object')
+    suite = Suite(sdf_name=sdf)
+    run_env.logger.info(f'Reading sdf {sdf} and populating Suite object')
+    success = suite.parse(make_call_tree=True)
+    if not success:
+        raise Exception(f'Parsing suite definition file {sdf} failed.')
+    run_env.logger.info(f'Successfully read sdf {suite.sdf_name}')
+    return suite
+
+def create_metadata_filename_dict(metapath):
+    """Given a path, read all .meta files in that directory and add them to a dictionary: the keys
+       are the name of the scheme, and the values are the filename of the .meta file associated
+       with that scheme"""
+
+    metadata_dict = {}
+    scheme_filenames=glob.glob(os.path.join(metapath, "*.meta"))
+    if not scheme_filenames:
+        raise Exception(f'No files found in {metapath} with ".meta" extension')
+
+    for scheme_fn in scheme_filenames:
+        schemes=find_scheme_names(scheme_fn)
+        # The above returns a list of schemes in each filename, but
+        # we want a dictionary of schemes associated with filenames:
+        for scheme in schemes:
+            metadata_dict[scheme]=scheme_fn
+
+    return metadata_dict
+
+
+def create_var_graph(suite, var, config, metapath, run_env):
+    """Given a suite, variable name, a 'config' dictionary, and a path to .meta files:
+         1. Creates a dictionary associating schemes with their .meta files
+         2. Loops through the call tree of the provided suite
+         3. For each scheme, reads .meta file for said scheme, checks for variable within that
+            scheme, and if it exists, adds an entry to a list of tuples, where each tuple includes
+            the name of the scheme and the intent of the variable within that scheme"""
+
+    # Create a list of tuples that will hold the in/out information for each scheme
+    var_graph=[]
+
+    run_env.logger.debug(f"reading .meta files in path:\n {metapath}")
+    metadata_dict=create_metadata_filename_dict(metapath)
+
+    run_env.logger.debug(f"reading metadata files for schemes defined in config file: "
+                  f"{config['scheme_files']}")
+
+    # Loop through call tree, find matching filename for scheme via dictionary schemes_in_files,
+    # then parse that metadata file to find variable info
+    partial_matches = {}
```

Comment:
Do you have examples of partial matches? How does tracking them help?

Reply:
This case is for when a user inputs something like "latent_heat" as their variable, which matches multiple standard names (e.g. `latent_heat_of_vaporization_of_water_at_0c`, `surface_upward_potential_latent_heat_flux`, etc.). This makes it easier for users who might not know the exact standard name of the variable they are looking for, or for something like, for example, any variable containing the word "temperature".

The final section in my PR message ("Partial match for variable") gives an example of this.